internal/jobservice/repository/postgres.go

```diff
@@ -330,3 +316,26 @@ func (s *JSRepoPostgres) GetSubscribedJobSets(ctx context.Context) ([]Subscribed
 	}
 	return tuples, nil
 }
+
+func (s *JSRepoPostgres) purgeExpired(ctx context.Context) {
+	jobsetsStmt := fmt.Sprintf(`DELETE FROM jobsets WHERE Timestamp < (extract(epoch from now()) - %d);`, s.jobServiceConfig.PurgeJobSetTime)
```

Comment:
`jobsetsStmt` and `jobServiceStmt` are assigned outside of the loop below. This means that every `PurgeJobSetTime` * seconds, this routine will re-run the same statements and never increment the time that is looked back on. In practice, this will mean that after the first successful run `purgeExpired` will effectively do nothing.